### ISSUE TITLE
Add a suite of "Simple*" kernels

### ISSUE THREAD

**Issue by dschwen** [idaholab/moose#6910]
### Description of the enhancement or error report

Based on feedback received by @amjokisaari I propose we add a suite of `Simple*` (such as `SimpleACInterface`) kernels that make assumption about the constness of mobilities and gradient interface parameters. 

We should make it very clear that these kernels are _only_ valid if certain documented conditions are fulfilled.

### Rationale for the enhancement or information for reproducing the error

Demonstrate that writing code is still easy for simple models. The general objects seem to be intimidating...

### Identified impact

Adds objects mostly for demonstration purposes.

**Comment by amjokisaari** [comment]
@dschwen, really cool of you to consider my feedback.  I think these "Simple" pieces will make it easier for people who want to see how the code works on the inside to get started (if they want to write their own stuff eventually, or are just curious, etc.).  I think this would lower the barrier to entry for new developers.

**Comment by friedmud** [comment]
:+1: for this.

This is why I am always against "generalizing" the "simple" objects in MOOSE.  Remember the first principle of MOOSE: Easy things should be simple.

(In case you're wondering... the second principle of MOOSE is that "Hard things should be possible" :-)

**Comment by amjokisaari** [comment]
This will also really be complimentary to the phase field benchmark
problems that we are developing with NIST.

**Comment by dschwen** [comment]
@friedmud I appreciate that principle, which is why I was happy to open the issue. I will say however that until a year ago we had very basic kernels that implemented the _wrong_ physics for all but the simplest cases. Any mobility or kappa that was not constant across the domain (and that is deceptively easy to do with moose material properties) would cause the physics to be plain wrong. 

We might need to have meta data on material properties to allow checks for constness...

**Comment by friedmud** [comment]
@dschwen Instead of checking for constness... I would really like to see a ConstMaterial... it would only be asked to compute its material properties _once_ (at the beginning of the simulation) and NEVER again.  In addition to allowing more checks... it would also be MUCH faster in the case where you really do have constant properties.

**Comment by dschwen** [comment]
Yes, absolutely. Quite a while ago I proposed to implement
GenericConstantMaterial that way.

**Comment by aeslaughter** [comment]
Should materials honor the "execute_on" option, be default they are set to
"initial linear nonlinear"? Then we don't need any special material class
type and we could also do stuff like "timestep_begin", etc.

**Comment by permcody** [comment]
I'll just remind you all that normal material properties aren't kept in memory anywhere so we'd have to make these types of properties stateful to pull this off. There's actually more work than that in that we don't have repartitioning with adaptivity so there would be even more work to be done before this was useful.

**Comment by friedmud** [comment]
A `ConstantMaterial` wouldn't need "storage".  It would just comoute into a material property that is of "max qps" size.., and that would get used on every element.

**Comment by permcody** [comment]
True, we'd need some framework adjustments though to handle arbitrary numbers of these simultaneously in the system.

**Comment by dschwen** [comment]
Would we? I think this almost works. The only issue are subdomain switches.

**Comment by friedmud** [comment]
I put a ticket in for creation of `ConstantMaterial`.  It would make a really nice addition to the framework.

**Comment by andrsd** [comment]
`MaterialData` is being swapped when we hit an element, so that stateful and normal material properties are behaving the same for users. So, this will be more complicated than it looks.

**Comment by friedmud** [comment]
It's swapped on a per-property basis - right?

**Comment by friedmud** [comment]
Oh - the discussion of the ConstantMaterial should go on over here: #6979

**Comment by dschwen** [comment]
This would be a good starter project for @Biyss! Let's talk about it.

**Comment by dschwen** [comment]
@amjokisaari let us know if you have further ideas for other simple kernels. Otherwise I'll close this issue for now.

**Comment by amjokisaari** [comment]
Thanks for pinging me :)  So, I'm trying to follow what was added.  It seems like various "simple" tests are in multiple directories within modules/phase_field/tests.  I think it would be nice if they were all in one subdirectory, like "Simple_examples" or something, so they can be found easily and used by a newb.  (I see @laagesen had a different thought, for a simple KKS test?)

What has been added at this point?  Is there other code that has been worked on to make it more generalized?  Those would be candidates for simple versions, as well.  

Finally, is there any documentation of these simple tests on the wiki page for the phase field module?  I looked briefly, but didn't see anything.  Since the point is to make it easy for new users to get started with these, having some documentation (mentioning their existence, their purpose, and where to find them) would be a great addition.

**Comment by GiudGiud** [comment]
Prefer Simple* or Example* ?
